static/js/util/IntegrationTestHelper.tsx

```diff
@@ -0,0 +1,156 @@
+import React from "react"
```

Comment:
This file, and its usage in `Header.test.tsx` is the biggest change in the PR.

Updating the publishing drawer alert broke some tests, so I used that as an opportunity to rewrite IntegrationTestHelper (or rather, create a new version). The biggest differences are:
- IntegrationTestHelper.tsx uses Jest for all mocking; the old file (now called integration_test_helper_old.tsx) used Sinon. Alice did a little work before she left toward replacing sinon mocks with Jest mocks in test files, but it was always a little awkward because the helper class used Sinon.
- IntegrationTestHelper.tsx uses `@testing-library/react`, whereas the old helper used `enzyme`.

See THIS ISSUE I'M GOING TO MAKE for why we should switch to @testing-library/react. The tldr is: enzyme doesn't seem to be actvitely developed (it does not support react 17 or react 18) and @testing-library/react is the officially-endorsed, canonical approach.